sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1057,25 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+class multinomial(CombinatorialFunction):
+    @classmethod
+    def eval(cls, *k):
+        from sympy.core import Add,Mul
+        k = [sympify(x) for x in k]
+        if all(x.is_Integer for x in k):
+            return factorial(Add(*k))/Mul(*(factorial(x) for x in k))
+        else:
+            from sympy import gamma, Add, Mul
+            return gamma(Add(*k) + 1)/Mul(*(gamma(x + 1) for x in k))
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        from sympy.core import Add,Mul
+        return factorial(Add(*args))/Mul(*(factorial(x) for x in args))
+
+    def _eval_rewrite_as_binomial(self, *args, **kwargs):
+        from sympy import binomial, Add, Mul
+        Mul(*[binomial(j,k) for j,k in zip([Add(*args[:i]) for i in range(1,len(args)+1)],args)])
+
+    def _eval_rewrite_as_gamma(self, *k, **kwargs):
+        from sympy import gamma, Add, Mul
+        return gamma(Add(*k) + 1)/Mul(*(gamma(x + 1) for x in k))
```

Comment:
`Add.fromiter(k)`?

Reply:
yeah ... it works too 😄

Reply:
can you hold on a sec ... I'll commit the code for the class ... and could you review that ?